Documentation/Functions/Typolink.rst

```diff
@@ -543,27 +543,30 @@ userFunc
     :name: typolink-userFunc
     :type: :ref:`data-type-function-name`
 
-    This passes the link-data compiled by the typolink function to a user-
-    defined function for final manipulation.
+    This passes the link-data compiled by the :typoscript:`typolink` function
+    to a user-defined PHP function for final manipulation.
 
     The :php:`$content` variable passed to the user-function (first parameter) is
     an array with the keys "TYPE", "TAG", "url", "targetParams" and
     "aTagParams".
 
-    TYPE is an indication of link-kind: mailto, url, file, page
+    `TYPE` is an indication of link-kind: `mailto`, `url`, `file`, `page`
 
-    TAG is the full <A>-tag as generated and ready from the typolink
+    `TAG` is the full `<a>`-tag as generated and processed from the :typoscript:`typolink`
     function.
 
-    The actual tag value is constructed like this:
+    The actual tag value must be returned with an instance of :php:`LinkResultInterface`,
```

Comment:
The userfunc will retrieve the pre-processed LinkResult allready. So, the userfunc is just a modificator of the delivered LinkResult which can be returns 1 by 1 or in a modified way.

PS: I don't want to explain how to build up your own LinkResult. Sure, it is possible, but IMO that should NOT be the way to go. It could leed to an unexpected email-link, because the TYPO3 delivered LinkResult was of type INPAGE. Brrr...please, don't do so.